File: lab3/s1/demo/remote/src/main/java/ru/uvuv643/ejb/remote/dto/human/request/CreateHumanBeingRequest.java
package ru.uvuv643.ejb.remote.dto.human.request;

import jakarta.validation.Valid;
import ru.uvuv643.ejb.remote.dto.enums.MoodDto;
import ru.uvuv643.ejb.remote.dto.enums.WeaponTypeDto;
import ru.uvuv643.ejb.remote.dto.human.CarDto;
import ru.uvuv643.ejb.remote.dto.human.CoordinatesDto;

import jakarta.validation.constraints.*;
import  jakarta.xml.bind.annotation.XmlAccessType;
import  jakarta.xml.bind.annotation.XmlAccessorType;
import  jakarta.xml.bind.annotation.XmlAttribute;
import  jakarta.xml.bind.annotation.XmlElement;
import  jakarta.xml.bind.annotation.XmlRootElement;
import java.io.Serializable;

@XmlRootElement(name = "CreateHumanBeingRequest")
@XmlAccessorType(XmlAccessType.FIELD)
public class CreateHumanBeingRequest implements Serializable {

        @XmlElement(name = "name")
        @NotBlank
        private String name;

        @XmlElement(name = "coordinates")
        @NotNull
        @Valid
        private CoordinatesDto coordinates;

        @XmlAttribute(name = "realHero")
        @NotNull
        private Boolean realHero;

        @XmlAttribute(name = "hasToothpick")
        @NotNull
        private Boolean hasToothpick;

        @XmlElement(name = "impactSpeed")
        @NotNull(message = "is not correct value")
        @Max(676)
        private Float impactSpeed;

        @XmlElement(name = "minutesOfWaiting")
        @NotNull(message = "is not correct value")
        @PositiveOrZero
        private Double minutesOfWaiting;

        @XmlElement(name = "car")
        @Valid
        @NotNull
        private CarDto car;

        @XmlElement(name = "mood")
        @NotNull
        @Valid
        private MoodDto mood;

        @XmlElement(name = "weaponType")
        @Valid
        private WeaponTypeDto weaponType;

        public CreateHumanBeingRequest() {
                this(null, null, false, false, null, null, null, null, null);
        }

        public CreateHumanBeingRequest(String name, CoordinatesDto coordinates, Boolean realHero, Boolean hasToothpick,
                                       Float impactSpeed, Double minutesOfWaiting, CarDto car, MoodDto mood,
                                       WeaponTypeDto weaponType) {
                this.name = name;
                this.coordinates = coordinates;
                this.realHero = realHero;
                this.hasToothpick = hasToothpick;
                this.impactSpeed = impactSpeed;
                this.minutesOfWaiting = minutesOfWaiting;
                this.car = car;
                this.mood = mood;
                this.weaponType = weaponType;
        }

        public String getName() {
                return name;
        }

        public void setName(String name) {
                this.name = name;
        }

        public CoordinatesDto getCoordinates() {
                return coordinates;
        }

        public void setCoordinates(CoordinatesDto coordinates) {
                this.coordinates = coordinates;
        }

        public Boolean getRealHero() {
                return realHero;
        }

        public void setRealHero(Boolean realHero) {
                this.realHero = realHero;
        }

        public Boolean getHasToothpick() {
                return hasToothpick;
        }

        public void setHasToothpick(Boolean hasToothpick) {
                this.hasToothpick = hasToothpick;
        }

        public Float getImpactSpeed() {
                return impactSpeed;
        }

        public void setImpactSpeed(Float impactSpeed) {
                this.impactSpeed = impactSpeed;
        }

        public Double getMinutesOfWaiting() {
                return minutesOfWaiting;
        }

        public void setMinutesOfWaiting(Double minutesOfWaiting) {
                this.minutesOfWaiting = minutesOfWaiting;
        }

        public CarDto getCar() {
                return car;
        }

        public void setCar(CarDto car) {
                this.car = car;
        }

        public MoodDto getMood() {
                return mood;
        }

        public void setMood(MoodDto mood) {
                this.mood = mood;
        }

        public WeaponTypeDto getWeaponType() {
                return weaponType;
        }

        public void setWeaponType(WeaponTypeDto weaponType) {
                this.weaponType = weaponType;
        }
}

File: lab2/s1/src/main/kotlin/ru/uvuv643/soa/api/v1/dto/human/request/CreateHumanBeingRequest.kt
package ru.uvuv643.soa.api.v1.dto.human.request

import EnumValue
import jakarta.validation.Valid
import jakarta.validation.constraints.*
import jakarta.xml.bind.annotation.XmlRootElement
import jakarta.xml.bind.annotation.XmlAccessorType
import jakarta.xml.bind.annotation.XmlAccessType
import jakarta.xml.bind.annotation.XmlAttribute
import jakarta.xml.bind.annotation.XmlElement
import ru.uvuv643.soa.api.v1.dto.enums.MoodDto
import ru.uvuv643.soa.api.v1.dto.enums.WeaponTypeDto
import ru.uvuv643.soa.api.v1.dto.human.CarDto
import ru.uvuv643.soa.api.v1.dto.human.CoordinatesDto

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
data class CreateHumanBeingRequest (

        @XmlElement(name = "name")
        @field:NotBlank
        val name: String?,

        @XmlElement(name = "coordinates")
        @field:NotNull
        @field:Valid
        val coordinates: CoordinatesDto?,

        @XmlAttribute(name = "realHero")
        @field:NotNull
        val realHero: Boolean?,

        @XmlAttribute(name = "hasToothpick")
        @field:NotNull
        val hasToothpick: Boolean?,

        @XmlElement(name = "impactSpeed")
        @field:NotNull(message = "is not correct value")
        @field:Max(676)
        val impactSpeed: Float?,

        @XmlElement(name = "minutesOfWaiting")
        @field:NotNull(message = "is not correct value")
        @field:PositiveOrZero
        val minutesOfWaiting: Double?,

        @XmlElement(name = "car")
        @field:Valid
        @field:NotNull
        val car: CarDto?,

        @XmlElement(name = "mood")
        @field:NotNull
        @field:Valid
        @field:EnumValue(
                enumClass = MoodDto::class,
        )
        val mood: String?,

        @XmlElement(name = "weaponType")
        @field:Valid
        @field:EnumValue(
                enumClass = WeaponTypeDto::class,
        )
        val weaponType: String?
)
{
        constructor() : this(
                name = null,
                coordinates = null,
                realHero = false,
                hasToothpick = false,
                impactSpeed = null,
                minutesOfWaiting = null,
                car = null,
                mood = null,
                weaponType = null
        )
}